docs/overview.rst

```diff
@@ -86,70 +94,255 @@ Important services in the diagram:
+incoming crash and does several things to it:
+
+1. assigns it a unique crash id
+2. tags it with a time stamp
+3. figures out whether the pipeline should process this crash or not
+
+The collector returns the crash id to the crash reporter which records it on
+the user's machine.
+
+The collector saves the crash data to Amazon S3 as a *raw crash* in a directory
+structure like this:
+
+.. code-block:: text
+
+   v2/
+     raw_crash/
+       000/
+         20160513/
+           00007bd0-2d1c-4865-af09-80bc02160513    raw crash metadata
+   v1/
+     dump_names/
+       00007bd0-2d1c-4865-af09-80bc02160513        list of minidumps for this crash
+     dump/
+       00007bd0-2d1c-4865-af09-80bc02160513        minidump file
+
+
+A crash id looks like this::
+
+  de1bb258-cbbf-4589-a673-34f800160918
+                               ^^^^^^^
+                               ||____|
+                               |  yymmdd
+                               |
+                               throttle result instruction
+
+
+.. seealso::
+
+   **Code**
+     https://github.com/mozilla-services/antenna
+
+   **Documentation**
+     https://antenna.readthedocs.io/
+
+   **Crash ids specification**
+     https://antenna.readthedocs.io/en/latest/spec_v1.html#crash-ids
+
+   **Socorro collector documentation**
+    :ref:`collector-chapter`
+
+
+Queued for processing by Pigeon (Python, AWS Lambda)
+----------------------------------------------------
+
+When the raw crash is saved to Amazon S3, Pigeon is invoked with an S3
+``ObjectCreated:Put`` event with the filename for the raw crash. The filename
+contains the crash id. Pigeon looks at the throttle result instruction character
+in the crash id to determine if the crash was deferred or accepted for
+processing.
+
+If the crash is not accepted for processing, then its story ends here. *[EXEUNT
+STAGE LEFT.]*
+
+If the crash is accepted for processing, Pigeon adds the crash id to the
+``socorro.normal`` processing queue in RabbitMQ.
+
+
+.. seealso::
+
+   **Code**
+     https://github.com/mozilla-services/socorro-pigeon
+
+   **Documentation**
+     https://github.com/mozilla-services/socorro-pigeon/blob/master/README.rst
+
+
+Processed by Processor (Python, Configman)
+------------------------------------------
+
+The processor gets a crash id from the ``socorro.normal`` queue in RabbitMQ.
+It fetches the raw crash data and related minidumps from Amazon S3.
+
+It passes all that information through the processing pipeline which consists of
+a series of rules that transform the crash into a processed crash.
+
+One of the rules runs the minidump-stackwalker over the minidump to extract
+information about the process and symbolicates the symbols on the stack. It also
+determines some other things about the state of the process when Firefox
+crashed.
+
+Another rule generates a crash signature from the stack of the crashind thread.
+We use crash signatures to bucket crashes so we can look at them as groups.
```

Comment:
Bucket is a little vague here. Maybe something like: "We use crash signatures to group together crashes that appear similar and were potentially caused by the same bug."